153_save_sensor_data/sensor_data_store.py

```diff
@@ -0,0 +1,238 @@
+           sensor_data[-1]['seconds'] > interpolated_time + desired_time_step):
+            # Get the smallest pair without going under
+            for index, data_set in enumerate(reversed(sensor_data)):
+                if data_set['seconds'] > interpolated_time + desired_time_step:
+                    smallest_set_without_going_under = data_set
+                if data_set['seconds'] < interpolated_time + desired_time_step:
+                    break
+
+            #Get the largest pair without going over
+            largest_set_without_going_over = sensor_data[-(index + 1)]
+            # Interpolate to find the new value
+            new_time = interpolated_time + desired_time_step
+            interpolated_set = interpolate_linear(
+                new_time,
+                # Time and values of the largest pair
+                largest_set_without_going_over,
+                # Time and values of the smallest pair
+                smallest_set_without_going_under
+            )
+            interpolated_data.append(interpolated_set)
+            if debugInterpolatedData: # print new data to screen to debug sensor
+                print(f"Time: {interpolated_set['seconds']:<10.2f} "
+                      f"CO2: {interpolated_set['co2']:>6.1f} ppm    "
+                      f"T: {interpolated_set['temp']:<3.2f}°C    "
+                      f"Humidity: {interpolated_set['humidity']:<3.2f}%")
+            interpolated_time += desired_time_step
+        
+        # If more than WRITE_QTY data points have been gathered, write to file
+        if len(interpolated_data) > WRITE_QTY:
+
+            column_names = ['time', 'co2','temp','humidity']
+            file_name = f"scd_data/raw/raw_data{interpolated_time}.csv"
+            #Write Raw Data
+            with open(file_name, 'w', newline='') as csvfile:
+                csv_writer = csv.DictWriter(csvfile, sensor_data[0].keys())
+                csv_writer.writeheader()
+                csv_writer.writerows(sensor_data)
+            # Write Interpolated Data
+            file_name = f"scd_data/interpolated/interpolated_data{interpolated_time}.csv"
+            with open(file_name, 'w', newline='') as csvfile:
+                csv_writer = csv.DictWriter(csvfile, interpolated_data[0].keys())
+                csv_writer.writeheader()
+                csv_writer.writerows(interpolated_data)
+
+            # Reset raw and interpolated data
+            # Remove all the data points smaller than the interpolated time
+            # Leaving the extra for the next interpolation.
+            total_vals = len(sensor_data)
+            i = total_vals - 1
+            while i >= 0 :
+                if sensor_data[i]['seconds'] < interpolated_time:
+                    sensor_data.remove(sensor_data[i])
+                i -= 1
+            # Just like in Java, this doesn't work in Python:
+            # for i, data_set in enumerate(sensor_data): # Keep some data for interpolation
```

Comment:
You can't edit an object while you are iterating on it.  This is generally solved by iterating over a copy of the object:
```suggestion
            # for i, data_set in enumerate(sensor_data[:]): # Keep some data for interpolation
```
(You can also use `list(sensor_data)` instead of `sensor_data[:]`).